sort/merge_sort.go

```diff
@@ -0,0 +1,34 @@
+package sort
+
+func mergeSort(arr []int) []int {
```

Comment:
### Security Issues

There are no direct security issues in this code snippet as it deals with basic sorting operations. However, if this function were to be used in a larger application, ensure that it is not exposed to untrusted input directly without proper validation and sanitization.